### ISSUE TITLE
Option to define custom handlebars helpers

### ISSUE THREAD

**Issue by Aokoooooo** [mockoon/mockoon#445]
**Is your feature request related to a problem? Please describe.**
To simplify the template JSON or to enhance the ability to process the data(like header or query)

**Describe the solution you'd like**
Maybe we could add a entry to write some JS code to define the helper we want.
In this JS file, we can access the base helper function, like `faker`, and the base data, like `header` of the request.
Then we could write some functions to do what we want.
After that, we export the functions or we registered the helpers manually.
Finally, we could use them in the template.

**Comment by rajshreeee** [comment]
Any lead on this?

**Comment by 255kb** [comment]
@rajshreeee This is not planned yet as it would probably require a plugin system. Plugins is something we might want to tackle in the future but it's not on the list of priorities (yet).

**Comment by chandima** [comment]
Wouldn't dependency injection solution be easier to implement? Where an instance of Handlebars with bound custom handlers is injected when Mockoon serverless is instantiated?

Like:
``javascript
const handlebarsInstance  = Handlebars.create();
handlebarsInstance.registerHelper("foo", function() {});

const mockoonServerless = new mockoon.MockoonServerless(mockEnv, handlebarsInstance);
``

**Comment by 255kb** [comment]
@chandima The idea is to have the same features in the desktop, CLI and serverless package. Having the possibility to reference a helper in a template that would be only available during the execution in the CLI or the serverless package is not something I want to start.

**Comment by adrian-green** [comment]
I agree that allowing the registration of custom helpers would be an enormously useful feature, taking things to a new level and _simplifying_ some types of data generation hugely. I came here because I thought I must have been doing something wrong - but no, it's just not implemented.

**Comment by 255kb** [comment]
Considering where the project is headed, I don't think this has a chance to be implemented in the future. Multiple reasons:

- the amount of work required. 
- our will to have the same set of features in desktop/CLI/libs/Cloud make it even harder to implement. 
- the security implications. 

I would favor adding more templating helpers to make life easier as they are easier to secure, and mostly harmless in terms of UI cluttering.
